pkg/proxy/runner/bounded_frequency_runner.go

```diff
@@ -136,18 +131,17 @@ func (bfr *BoundedFrequencyRunner) Loop(stop <-chan struct{}) {
 			bfr.minIntervalTimer.Stop()
 			bfr.maxIntervalTimer.Stop()
 
-			// do not allow to reset the retryInterval during the execution
-			bfr.retryMu.Lock()
-			defer bfr.retryMu.Unlock()
-			// stop but do not reset it, it is a one shot timer,
-			// only rearmed when calling Retry().
-			bfr.retryIntervalTimer.Stop()
-			bfr.retrying = false
-
 			// avoid crashing if the function executed crashes
 			defer utilruntime.HandleCrash()
-			bfr.fn()
-
+			err := bfr.fn()
+			// an error will schedule a retry after the retryInterval,
+			// any successful run until that period will stop the retry attempt.
+			if err != nil {
+				klog.V(3).InfoS("retrying", "name", bfr.name, "interval", bfr.retryInterval)
+				bfr.retryIntervalTimer.Reset(bfr.retryInterval)
+			} else {
+				bfr.retryIntervalTimer.Stop()
```

Comment:
You shouldn't need a separate `retryIntervalTimer`; you can just reset `maxIntervalTimer` to either `retryInterval` or `maxInterval` depending on whether there was an error

Reply:
good point

Reply:
fixed in next commit